### ISSUE TITLE
Implement summation logic for matched (M) and extended (E) grant codes 

### ISSUE THREAD

**Issue by daissatou2** [OpenFn/akf#6]
## Background, context, and business value
See #2 for project background.

## The specific request, in as few words as possible
All `Grant Codes` in the BC actuals CSV file will end with a number _**except**_  in certain cases where the `Grant Code` will end with either the letter "M" or the letter "E".  For these grant codes with the trailing "M" or "E", treat it as the grant code without the trailing letter. 

For example, in the BC actuals export we expect to find `Grant Code` that equals `ECHO496`, `ECHO496M`, and `ECHO496E`. **All three of these grant codes should be treated as `ECHO496`.** I.e. when we aggregate by grant code for `ECHO496` we should also include the `Sum of Actual` for `ECHO496M` and `ECHO496E` rows. There is no `ECHO496M` or `ECHO496E` in Salesforce, only `ECHO496`. 

### Expected data volumes
TBD - but we expect CSVs to have sometimes 20k-100k rows

## state.json

Job 1:
```json
{
  "configuration": ["SEE LAST PASS: AKF Sharepoint Site (FYI MFA token is sent to Aicha's phone'"],
  "data": "NA"
  "cursor": "2023-07-30"
}
```

Job 2:
```json
{
  "configuration": ["NONE"],
  "data": "Use data passed from job 1"
  
}
```
Job 3:
```json
{
  "configuration": ["SEE LAST PASS: Sandbox - Partial 2023]",
  "data": "Use data passed from job 2"
  
}
```

### job name
applyMappings.js

### adaptor
http

### trigger
Flow

### output/testing
To test this, use the file named `bc_actuals_export_2023_08_10.csv` (with the files currently in the `BC Actuals Exports` folder and cursor date of `2023-07-30`, this file should be returned from the GET job.) 

The expected output for this file is shown in the "AGGREGATED FINANCIAL LINES TO MAP TO SALESFORCE DATA MODEL..." table in the " Aggregated output V2" tab of this [workbook](https://docs.google.com/spreadsheets/d/1k1H1sTYTk3hcX5UtD9Te6Y1bxAsZL0de/edit#gid=566755775). The rows I highlighted yellow are rows that include summations from grant codes with the "M" or "E" or both. If you take a look at the "Aggregated output V1" tab (what was initially implemented), you'll see how those rows highlighted yellow are now being summed with the grant code without the training letter. 

In summary, we should send 27 records to Salesforce with this file. The `SUM of Sum of Actual` column tells you the `Amount_Actual_Local_Currency__c` value that should be sent for each of the rows. 

## Toggl 
AKF BC M3 - Integration Setup

**Comment by daissatou2** [comment]
@mtuchi we tested this one together a few weeks ago but it looks like some recent changes may have impacted this implementation. See this run [log](https://www.openfn.org/projects/akf-bc-aims-integration/runs/064dfa6e-c654-76c5-8a1a-13c7edc64728). All of the failures are because the job is trying to find a budget with an id that includes the "m" or the "e". For example, on line 32 the error message says:

  `    "INVALID_FIELD:Foreign key external ID: 1ec10m_default_2022_afghanistan akf not found for field External_Id__c in entity ampi__Budget__c:--"`

The id that we are using to find the budget should be `1ec10_default_2022_afghanistan akf ` not `1ec10m_default_2022_afghanistan akf `. The job should remove the trailing "e" or the "m" and treat it as just "1ec10".

You can replicate this by clearing the state and testing with the `bc_actuals_2023_07_14.csv` file on the AKF SharePoint site.

**Comment by mtuchi** [comment]
@daissatou2 i have fixed the bug that was causing the externalId to have M or E, here #12. You can proceed testing 
Cc @aleksa-krolls

**Comment by aleksa-krolls** [comment]
@daissatou2 please close if resolved and mapping specs/workflow diagrams all up-to-date.
